### ISSUE TITLE
Python packages detection error.

### ISSUE THREAD

**Issue by saganshul** [owtf/owtf#748]
There are some python packages which install on running installation script but not getting detected.
## Expected Behavior

It should install all the dependencies and detect them properly.
## Current Behavior

The following python libraries seem missing : 
   Markdown,pyOpenSSL,SQLAlchemy
Haven't you run the install script? 
   python2 install/install.py

If you are sure you ran the install script, install the missing libraries separately
 pip install --upgrade -r install/owtf.pip

But running "$pip install --upgrade -r install/owtf.pip" gives still same error
## Possible Solution

Bug in: owtf/framework/dependency_check.py
## Logs

[_] Running following command
sudo -E pip2 install --upgrade -r /root/owtf/install/owtf.pip
Requirement already up-to-date: cookies==2.2.1 in /usr/local/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 6))
Requirement already up-to-date: ipaddr==2.1.11 in /usr/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 7))
Requirement already up-to-date: lxml==3.6.4 in /usr/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 8))
Requirement already up-to-date: Markdown==2.6.6 in /usr/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 9))
Requirement already up-to-date: pexpect==4.2.1 in /usr/local/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 10))
Requirement already up-to-date: psycopg2==2.6.2 in /usr/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 11))
Requirement already up-to-date: PTP==0.3.0 in /usr/local/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 12))
Requirement already up-to-date: pycurl==7.43.0 in /usr/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 13))
Requirement already up-to-date: six==1.10.0 in /usr/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 15))
Requirement already up-to-date: pyOpenSSL==16.1.0 in /usr/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 16))
Requirement already up-to-date: python-owasp-zap-v2==0.0.9 in /usr/local/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 17))
Requirement already up-to-date: PyVirtualDisplay==0.2 in /usr/local/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 18))
Requirement already up-to-date: rdflib==4.2.1 in /usr/local/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 19))
Requirement already up-to-date: selenium==2.53.6 in /usr/local/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 20))
Requirement already up-to-date: SQLAlchemy==1.0.15 in /usr/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 21))
Requirement already up-to-date: tornado==4.4.1 in /usr/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 22))
Requirement already up-to-date: requests==2.10.0 in /usr/local/lib/python2.7/dist-packages (from -r /root/owtf/install/owtf.pip (line 23))
Requirement already up-to-date: ptyprocess>=0.5 in /usr/lib/python2.7/dist-packages (from pexpect==4.2.1->-r /root/owtf/install/owtf.pip (line 10))
Requirement already up-to-date: EasyProcess in /usr/local/lib/python2.7/dist-packages (from PyVirtualDisplay==0.2->-r /root/owtf/install/owtf.pip (line 18))
Requirement already up-to-date: pyparsing in /usr/lib/python2.7/dist-packages (from rdflib==4.2.1->-r /root/owtf/install/owtf.pip (line 19))
Requirement already up-to-date: isodate in /usr/local/lib/python2.7/dist-packages (from rdflib==4.2.1->-r /root/owtf/install/owtf.pip (line 19))
Requirement already up-to-date: SPARQLWrapper in /usr/local/lib/python2.7/dist-packages (from rdflib==4.2.1->-r /root/owtf/install/owtf.pip (line 19))
Requirement already up-to-date: html5lib in /usr/local/lib/python2.7/dist-packages (from rdflib==4.2.1->-r /root/owtf/install/owtf.pip (line 19))
Requirement already up-to-date: keepalive>=0.5 in /usr/local/lib/python2.7/dist-packages (from SPARQLWrapper->rdflib==4.2.1->-r /root/owtf/install/owtf.pip (line 19))
Requirement already up-to-date: setuptools>=18.5 in /usr/local/lib/python2.7/dist-packages (from html5lib->rdflib==4.2.1->-r /root/owtf/install/owtf.pip (line 19))
Requirement already up-to-date: webencodings in /usr/local/lib/python2.7/dist-packages (from html5lib->rdflib==4.2.1->-r /root/owtf/install/owtf.pip (line 19))
[_] Finished!
[*] Start OWTF by running './owtf.py' in parent directory
## Your Environment

- Operating System and version (like Kali, Debian, ArchLinux, etc): Kali2

**Comment by delta24** [comment]
This is due to Kali 2016.x and pip not playing nicely with each other :D
It is high time anyway that we shift to a virtualenv-based setup process.

**Comment by DarKnight24** [comment]
@delta24 Do we need any thing to be done over this issue?

**Comment by delta24** [comment]
@DarKnight24 We need to move the install process to use virtualenv and virtualenvwrapper by default, so that these issues do not pop up in future.

**Comment by jcmoreno** [comment]
Hi, I'm running into this exact same issue but just with Markdown. See below:

`root@kali:/opt/owtf# apt-get install markdown
Reading package lists... Done
Building dependency tree       
Reading state information... Done
markdown is already the newest version (1.0.1-9).
0 upgraded, 0 newly installed, 0 to remove and 1155 not upgraded.
root@kali:/opt/owtf# pip install Markdown
Requirement already satisfied: Markdown in /usr/lib/python2.7/dist-packages
root@kali:/opt/owtf# ./owtf.py 
The following python libraries seem missing : 
	Markdown

Haven't you run the install script? 
	python2 install/install.py

If you are sure you ran the install script, install the missing libraries seperately
	pip install --upgrade -r install/owtf.pip
`

**Comment by delta24** [comment]
@jcmoreno did you try using virtualenv?

**Comment by jcmoreno** [comment]
I don't think so. I just followed the documentation and ran the following:
```
git clone https://github.com/owtf/owtf.git
cd owtf/
./install/install.py
```

**Comment by delta24** [comment]
@jcmoreno You can install virtualenv by running

1. `pip install virtualenv`.
2. Create a virtual environment using `virtualenv <env name>`
3. Activate the environment - `source <env_name>/bin/activate`
4. Install OWTF

**Comment by jcmoreno** [comment]
I'll try that then. Thanks!

**Comment by DarKnight24** [comment]
There are some issues in installation.
* Why are we using pip command twice, see [here](https://github.com/owtf/owtf/blob/develop/install/distro-independent.cfg#L13) and [here](https://github.com/owtf/owtf/blob/develop/install/owtf.pip). Both the command have pyopenssl in it.
* Debian installation script also install pip, which is already installed, see [here](https://github.com/owtf/owtf/blob/develop/install/distro-independent.cfg#L8) and [here](https://github.com/owtf/owtf/blob/develop/install/debian/install.sh#L25).

@delta24 @7a  have a look at it.

**Comment by delta24** [comment]
@DarKnight24 that seems like a good idea. We should move all `pip` packages to `owtf.pip` and install `pip` only once.

**Comment by rishi62** [comment]
Hi, i am facing this issue when installing owtf,currently i want to understand how owtf is working but this issue is not fixing. What i know is that this dependecies are installed but the tool is not detecting the correct path i guess. So it will be good if someone can clear me on how to fix this issue.

**Comment by DePierre** [comment]
@rishi62 It would be nice to have some information regarding your setup so that we could properly help you. For instance, could you tell us:

+ What OS are you using? And what distribution?
+ Could you create a pastebin containing the output of the boostrap.sh script?
+ Could you paste the error shown when start owtf.py?
+ Could you paste the pip packages installed on your system (using `pip freeze` iirc)

**Comment by rishi62** [comment]
Hey its working when i ran bootstrap.sh again i got a pip related error then i just reinstalled pip for python && python3 then the whole thing worked and now i am using the web interface of owtf on kali linux 2016.2.

**Comment by DePierre** [comment]
@rishi62 So is it working now?

**Comment by rishi62** [comment]
yup!

**Comment by jack17529** [comment]
So after renaming the bug is solved or not?

**Comment by DePierre** [comment]
@jack17529 a PR is currently being reviewed (see #801) to use virtualenv and limit the clash between Kali and pip. Feel free to contribute to the review with some comments/questions in the PR thread.

**Comment by jack17529** [comment]
I am  new sir can you tell me what is PR?
